### ISSUE TITLE
The dice sum of damage in error

### ISSUE THREAD

**Issue by ZolOnTheNet** [elfonochasis/Sistema-Hitos-Foundry-VTT#34]
Hello,
Sorry my English is not fluent, but i try to explain

when you entering a Damage formula, like "mM+3", if dice give 1,2,3 (m= 1, M = 3) the result damage is 13+3 = 16  because in your code, you write (dice.js, line 62 and chat.js line 41)
let damageBase = eval(weapon.damage.replace("m",lookup["m"]).replace("C",lookup["C"]).replace("M",lookup["M"]))
to quickly corret, you can change in that :
let damageBase = eval(weapon.damage.replace("m","+"+lookup["m"]).replace("C","+"+lookup["C"]).replace("M","+"+lookup["M"]))

It's allow code like mCM + n where n is a constant. By the way, in the rule damage formula in Hitos system can contain another dices, like CM+1D10 (noted CM+1D, but I want to integrate any dices in formula ie "M+1d4"). I wrote solution for that, but i want to know if you are interest by my solution. It's not complete because i don't find to integrate in roll into "Dice So Nice" roll (chat roll parameter)...

Thank for the good work of Hitos/FoundryVTT

In french :
Bonjour, 
J'essaie d'expliquer le problème : le calcul des dommages dans votre portage de Hitos donne un mauvais résultat : cela est du au fait que dans Hitos, il indique les dommages sous la forme mCM où chaque dé, en fonction de l'ordre, correspond au minimum, Central, Maximum respectivement. le codage mCM n'indique pas une concaténation des valeurs mais une addition qu'il faut interpréter par m+C+M, or votre code colle les correspondance, ainsi pour le résultat des dés  1,2,3 (m = 1, C = 2, M=3), la formule "mM+3" donne 16 (13+3), au lieu de 7.

Une façon de corriger rapidement la chose est de replacer les lignes 62 de dice.js et 41 de chat.js : 
let damageBase = eval(weapon.damage.replace("m",lookup["m"]).replace("C",lookup["C"]).replace("M",lookup["M"]))
par
let damageBase = eval(weapon.damage.replace("m","+"+lookup["m"]).replace("C","+"+lookup["C"]).replace("M","+"+lookup["M"]))
ce qui ajoute les "+" entre chaque élement de la formule

La contrainte est qu'il faut utiliser la forme "mCM+n" et pas "n+mCM". D'ailleurs, dans le système Hitos, il  y a la possibilité d'ajouter des dés à la formule, normalement noté 1D pour 1D10. J'ai fait une modif, pas tout a fait satisfaisante qui permet d'ajouter des des (quelque soit la formule : +1d6, +1d10 etc...). Je n'ai pas encore trouver le moyen de le rajouter à la visualisation des lancers de dés de "Dice So Nice"... Je veux bien vous la fournir, s'il vous le souhaiter...

Cordialement, 
Et encore merci pour se portage de Hitos sous FoundryVtt, bon travail...

**Comment by HonzoNebro** [comment]
This is correct, there is an issue on attack rolls, and the proposed solution works.

Also, give us permission to create PR, there are several issues

**Comment by elfonochasis** [comment]
Hello!
Nice work and thanks for your feedback! You can make a fork and PR or you can upload solution in a zip here.

**Comment by elfonochasis** [comment]
I've uploaded version 1.1.2 with fixed damage. @ZolOnTheNet, could you share with me the another version with dices as part of the formula?

**Comment by ZolOnTheNet** [comment]
Hello ! 
I give you chat.js and dice.js, but i not fully happy with this code : 
I begin in js : i think they are more effecient coding
I not happy on the propagation of dice roll, if player use reroll (drama action), the damage wil be reroll too.
I think to avoid this (by calculation)

in dice.js : my modification between  line 66 to 80 and in chat.js is line 42 to 54
The idea is to not evaluate immediately the expression, but fin if it have a "D" (or "d") to roll the expression

I'm sorry to not send you a pull, i begin in git and github too !
I hope that help you !

Thank for your job...
[Dice et chat.zip](https://github.com/elfonochasis/Sistema-Hitos-Foundry-VTT/files/10699080/Dice.et.chat.zip)

**Comment by elfonochasis** [comment]
Nice @ZolOnTheNet! I will check it and publish on next days.
Thanks for your colaboration!

**Comment by ZolOnTheNet** [comment]
Super ! I dont know to add roll to have on Dice So Nice ; perhaps to add manually to the first roll passed to the chat parameters... The two files was written with the 1.1.0 code (main branch), if you have made change after.
Have a good night

**Comment by ZolOnTheNet** [comment]
One more Note : in file dice.js, line 74 to 77, it's a try to calculate the damage roll (minus the Code mCM), it's experimental : the calculation need to be more verified. This value (only de roll bonus damage) can be transmit to the next Drama dialog to avoid to re roll bonus damage (for coherence between the first and the second roll).

I prefer to explain if you don't understand the reason of this part of the code...
